sdk/metric/view/view.go

```diff
@@ -17,22 +17,195 @@
+		desc.InstrumentName = v.viewName
+	}
+	if v.description != "" {
+		desc.InstrumentDescription = v.description
+	}
+	return desc, true
+}
+
+// TransformAttributes filters an attribute set to the keys in the config. If no
+// filter was provided the original set is returned.
+func (v View) TransformAttributes(input attribute.Set) attribute.Set {
+	if len(v.filterKeys) == 0 {
+		return input
+	}
+	out, _ := input.Filter(attribute.Filter(func(kv attribute.KeyValue) bool {
+		_, ok := v.filterKeys[kv.Key]
+		return ok
+	}))
+	return out
+}
+
+// TODO: Provide Transform* for AggregationKind (#2816)
+
+func (v View) matchName(name string) bool {
+	return v.instrumentName == nil || v.instrumentName.MatchString(name)
 }
 
-// TODO (#2837): add getter functions for all the internal fields of a Config.
+func (v View) matchLibraryName(name string) bool {
+	return v.scope.Name == "" || name == v.scope.Name
+}
+
+func (v View) matchLibraryVersion(version string) bool {
+	return v.scope.Version == "" || version == v.scope.Version
+}
+
+func (v View) matchLibrarySchemaURL(schemaURL string) bool {
+	return v.scope.SchemaURL == "" || schemaURL == v.scope.SchemaURL
+}
 
-// Option applies a configuration option value to a view Config.
+// Option applies a Configuration option value to a view Config. All options
+// will be used together to determine match and transforms.
 type Option interface {
-	apply(Config) Config
+	apply(View) View
+}
+
+type optionFunc func(View) View
+
+func (f optionFunc) apply(v View) View {
+	return f(v)
+}
+
+// MatchInstrumentName will match an instrument based on the instruments name.
+// This will accept wildcards of * for zero or more characters, and ? for
+// exactly one character. A name of "*" (default) will match all instruments.
+func MatchInstrumentName(name string) Option {
+	return optionFunc(func(v View) View {
+		if strings.ContainsAny(name, "*?") {
+			v.hasWildcard = true
+		}
+		name = regexp.QuoteMeta(name)
+		name = "^" + name + "$"
+		name = strings.ReplaceAll(name, "\\?", ".")
+		name = strings.ReplaceAll(name, "\\*", ".*")
+		v.instrumentName = regexp.MustCompile(name)
+		return v
+	})
 }
 
-// TODO (#2837): implement view match options.
-// TODO (#2837): implement view annotation options.
+// TODO: Implement when InstrumentKind and NumberKind are defined
+// func MatchInstrumentKind(k sdkinstrument.Kind) Option {
+// 	return optionFunc(func(v Config) Config {
+// 		v.instrumentKind = k
+// 		return v
+// 	})
+// }
+
+// func MatchNumberKind(k number.Kind) Option {
+// 	return optionFunc(func(v Config) Config {
+// 		v.numberKind = k
+// 		return v
+// 	})
+// }
 
-// New returns a new and configured view Config.
-func New(opts ...Option) Config {
-	return Config{}
+// MatchInstrumentationLibrary will do an exact match on any
+// instrumentation.Library field that is not blank (""). Default is to match all
+// InstrumentationLibraries.
+func MatchInstrumentationLibrary(lib instrumentation.Library) Option {
```

Comment:
The function ```MatchInstrumentationLibrary```'s body makes me think the name  ```WithInstrumentationLibrary``` is more suitable, just as following ```WithName```, ```WithDescription``` functions. 

Is this name ```MatchInstrumentationLibrary``` designed on purpose?
Is

Reply:
The view is a device that has two responsibilities:

1. To select instruments.
2. Modify selected instruments in some way.  
 
This option doesn't change the Instrumentation Library, it will only match instruments.  The specification doesn't say that we can change the library, so we don't have an option (or logic) that will do that kind of modification.